src/__tests__/components.test.js

```diff
@@ -0,0 +1,163 @@
+import { BrowserRouter, Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import { render, screen, fireEvent } from '@testing-library/react';
+import renderer from 'react-test-renderer';
+import { Provider } from 'react-redux';
+// import ReactTestUtils, { act } from 'react-dom/test-utils';
+import store from '../redux/configureStore';
+import { ThemeProvider } from '../contexts/theme';
+import AppRoutes from '../AppRoutes';
+import NotFound from '../features/NotFound/NotFound';
+import MainPage from '../pages/main/MainPage';
+import BeerPage from '../pages/beer/BeerPage';
+import ScrollToTop from '../features/ScrollToTop/ScrollToTop';
+// import App from '../App';
+
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: jest.fn().mockImplementation((query) => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // deprecated
+    removeListener: jest.fn(), // deprecated
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    dispatchEvent: jest.fn(),
+  })),
+});
+
+// Routing testing
+
+describe('Navbar renders', () => {
+  test('should redirect and update history', () => {
+    const history = createMemoryHistory();
+    render(
+      <Provider store={store}>
+        <ThemeProvider>
+          <Router location={history.location} navigator={history}>
+            <AppRoutes />
+          </Router>
+        </ThemeProvider>
+      </Provider>,
+    );
+
+    fireEvent.click(screen.getByTestId(/Beer Lovers/));
+    expect(history.location.pathname).toEqual('/');
+  });
```

Comment:
- [ ] You only added screenshot tests for your components, Please also add unit tests for some pure functions in Redux code and integration tests using React Testing Library mocking data fetching and Redux Store connection.

Reply:
https://github.com/manq2010/beer-lovers/blob/22948f770288591531dfc6c66110a675d722d6f0/src/__tests__/components.test.js#L1-L163

Please note the test file has all the required test